src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4041,6 +4045,33 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
             {
                 analyzedArguments.Free();
             }
+
+            static void validateRecordCopyConstructor(NamedTypeSymbol containingType, MethodSymbol constructor, NamedTypeSymbol baseType, MethodSymbol baseCtor, Location errorLocation, DiagnosticBag diagnostics)
+            {
+                // Are we dealing with a user-defined copy constructor in a record type?
+                if (containingType is SourceNamedTypeSymbol sourceType &&
+                    sourceType.IsRecord &&
+                    SynthesizedRecordCopyCtor.HasCopyConstructorSignature(constructor))
+                {
+                    if (baseType.SpecialType == SpecialType.System_Object)
+                    {
+                        return;
+                    }
+
+                    // Unless the base type is 'object', the constructor should invoke a base type copy constructor
+                    HashSet<DiagnosticInfo> ignoredUseSiteDiagnostics = null;
+                    var foundBaseCopyCtor = SynthesizedRecordCopyCtor.FindCopyConstructor(baseType, containingType, ref ignoredUseSiteDiagnostics);
+                    if (foundBaseCopyCtor is null)
```

Comment:
>if (foundBaseCopyCtor is null) [](start = 20, length = 30)

I do not understand why are we performing this check at all. It really doesn't matter whether there is a copy constructor in the base or not if we are bound to something else. I think the diagnostics is actually going to be confusing. Rather than saying, you are using wrong thing, we say, something is missing in the base. Consider removing this logic.

Reply:
I think we should always report ERR_CopyConstructorMustInvokeBaseCopyConstructor here, regardless of the situation.

---
In reply to: [439014453](https://github.com/dotnet/roslyn/pull/45022#discussion_r439014453) [](ancestors = 439014453)

Reply:
You're suggesting removing the base copy constructor symbol from `ERR_CopyConstructorMustInvokeBaseCopyConstructor`, then?

---
In reply to: [439015232](https://github.com/dotnet/roslyn/pull/45022#discussion_r439015232) [](ancestors = 439015232,439014453)

Reply:
> You're suggesting removing the base copy constructor symbol from ERR_CopyConstructorMustInvokeBaseCopyConstructor, then?

Yes, I think that would be fine.

---
In reply to: [439056805](https://github.com/dotnet/roslyn/pull/45022#discussion_r439056805) [](ancestors = 439056805,439015232,439014453)